### PR TITLE
Add `TypeAdapter` section to `concepts` page of docs

### LINKED ISSUE #8051: Add `TypeAdapter` section to concepts docs
There used to be a `TypeAdapter` section in the `Usage` docs, but we moved all of that content to the `API Documentation` section.

Let's add a section to the `Concepts` docs explaining how to use a `TypeAdapter`.

### PR DESCRIPTION
**Pull request by sydney-runkle**
## Change Summary

Moving more conceptual overview of `TypeAdapter` to `concepts` section of the docs

## Related issue number

Fix #8051

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [x] Tests pass on CI
* [x] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: docs/concepts/type_adapter.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+You may have types that are not `BaseModel`s that you want to validate data against.
+Or you may want to validate a `List[SomeModel]`, or dump it to JSON.
+
+??? api "API Documentation"
+    [`pydantic.type_adapter.TypeAdapter`][pydantic.type_adapter.TypeAdapter]<br>
+
+For use cases like this, Pydantic provides [`TypeAdapter`][pydantic.type_adapter.TypeAdapter],
+which can be used for type validation, serialization, and JSON schema generation without creating a
```
Comment (dmontagu) [inline docs/concepts/type_adapter.md:8]:
```suggestion
which can be used for validation, serialization, and JSON schema generation without needing to create a
```

File: docs/concepts/type_adapter.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+from typing_extensions import TypedDict
+
+from pydantic import TypeAdapter, ValidationError
+
+class User(TypedDict):
+    name: str
+    id: int
+
+UserListValidator = TypeAdapter(List[User])
```
Comment (dmontagu) [inline docs/concepts/type_adapter.md:26]:
```suggestion
UserListAdapter = TypeAdapter(List[User])
```
(and below).

I think better to use `Adapter` than `Validator` to help people realize it's able to do more than validation, also I just think it's a better naming convention and would want to push people toward using it as they read the docs, even if they are only doing validation with the value.

File: docs/concepts/type_adapter.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    1 validation error for list[typed-dict]
+    0.id
+      Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='wrong', input_type=str]
+    '''
+```
+
+Note:
+    Despite some overlap in use cases with [`RootModel`][pydantic.root_model.RootModel],
+    [`TypeAdapter`][pydantic.type_adapter.TypeAdapter] should not be used as a type annotation for
+    specifying fields of a `BaseModel`, etc.
```
Comment (Kludex) [inline docs/concepts/type_adapter.md:46]:
```suggestion
!!! note
    Despite some overlap in use cases with [`RootModel`][pydantic.root_model.RootModel],
    [`TypeAdapter`][pydantic.type_adapter.TypeAdapter] should not be used as a type annotation for
    specifying fields of a `BaseModel`, etc.
```

File: mkdocs.yml [hunk 4]
```diff
@@ -67,6 +67,7 @@ nav:
   - Dataclasses: concepts/dataclasses.md
   - Postponed Annotations: concepts/postponed_annotations.md
   - Strict Mode: concepts/strict_mode.md
+  - TypeAdapter: concepts/type_adapter.md
```
Comment (Kludex) [inline mkdocs.yml:70]: Maybe?
```suggestion
  - Type Adapter: concepts/type_adapter.md
```

File: docs/concepts/type_adapter.md (hunk 2 above)
Comment (sydney-runkle) [reply docs/concepts/type_adapter.md:28]: Agreed, thanks for the tip.

File: docs/concepts/type_adapter.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+# item_data = requests.get('https://my-api.com/items').json()
+item_data = [{'id': 1, 'name': 'My Item'}]
+
+items = TypeAdapter(List[Item]).validate_python(item_data)
+print(items)
+#> [Item(id=1, name='My Item')]
+```
+
+[`TypeAdapter`][pydantic.type_adapter.TypeAdapter] is capable of parsing data into any of the types Pydantic can
+handle as fields of a [`BaseModel`][pydantic.main.BaseModel].
```
Comment (dmontagu) [inline docs/concepts/type_adapter.md:81]:
```suggestion

!!! info "Performance considerations"
    When creating an instance of `TypeAdapter`, the provided type must be analyzed and converted into a pydantic-core
    schema. This comes with some non-trivial overhead, so it is recommended to create a `TypeAdapter` for a given type
    just once and reuse it in loops or other performance-critical code.
```
Not necessary to add in this PR, but I feel something like this belongs here.

**Review by dmontagu — APPROVED** [review]

File: docs/concepts/type_adapter.md (hunk 5 above)
Comment (sydney-runkle) [reply docs/concepts/type_adapter.md:81]: Awesome, I'll add that to this PR.

**Pull request merged**